### PR TITLE
GEOMESA-1508 Using exiting executor for query thread timeout reaper

### PR DESCRIPTION
**Pull request by elahrvivaz**
~~Note: stacks on top of #1184~~
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/ThreadManagement.scala [hunk 1]
```diff
@@ -53,9 +54,9 @@ object ThreadManagement extends LazyLogging {
     }
   }
 
-  private val executor = Executors.newSingleThreadScheduledExecutor()
+  private val executor = MoreExecutors.getExitingScheduledExecutorService(new ScheduledThreadPoolExecutor(1))
   executor.scheduleWithFixedDelay(reaper, interval, interval, TimeUnit.MILLISECONDS)
-  sys.addShutdownHook(executor.shutdown())
+  sys.addShutdownHook(executor.shutdownNow())
```
Comment (mdzimmerman) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/ThreadManagement.scala]: This resolves the hanging problem for the Accumulo QuickStart, both with and without a ``-Dgeomesa.query.timeout.millis=<value>`` argument.

Comment (jahhulbert-ccri) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/ThreadManagement.scala]: any reason to keep the shutdown hook here if using ```MoreExecutors.getExitingScheduledExecutorService``` ?

Comment (jahhulbert-ccri) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/ThreadManagement.scala]: i guess it doesn't call shutdown now... only shutdown

Comment (elahrvivaz) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/ThreadManagement.scala]: yeah, i didn't really look into this too closely but figure it couldn't hurt (and we do that with the other exiting executors too)

Comment (jahhulbert-ccri) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/ThreadManagement.scala]: good enough for me

**Pull request merged**